aws-api/src/main/java/com/amplifyframework/api/aws/GsonGraphQLResponseFactory.java

```diff
@@ -124,7 +125,11 @@
                 jsonData.getAsJsonObject().has(ITEMS_KEY)
         ) {
             Iterable<T> data = parseDataAsList(jsonData.getAsJsonObject().get(ITEMS_KEY), classToCast);
-            return new GraphQLResponse<>(data, errors);
+            String nextToken = null;
+            if(jsonData.getAsJsonObject().has(NEXT_TOKEN_KEY) && jsonData.getAsJsonObject().get(NEXT_TOKEN_KEY).isJsonPrimitive()) {
+                nextToken = jsonData.getAsJsonObject().get(NEXT_TOKEN_KEY).getAsString();
+            }
```

Comment:
`NEXT_TOKEN_KEY` is an AppSync detail that really ought not to be in this `GsonGraphQLResponseFactory`. However, so is `ITEMS_KEY`, right above. The API plugin sort of leaks AppSync stuff, here and there. There are probably bigger battles worth fighting. I'm more sensitive to us leaking this kind of stuff into _`core`_.